### PR TITLE
Cloud test connection string updates

### PR DESCRIPTION
**Pull request by MattGal**
Addresses https://github.com/dotnet/core-eng/issues/416.  Some minor bug fixes and cleanup that I incidentally noticed.

Note that I observed a fairly gnarly bug in ListAzureContainers that will require switching to using continuation tokens, but since noone's noticed it and it seems to be a fallback codepath I will just file it on core-eng.

@mmitche

### CODE REVIEW

**Comment by MattGal** [comment]
@chcosta  or @mmitche , can I get a signoff please?

**Comment by MattGal** [comment]
ping

File: src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs [hunk 1]
```diff
@@ -77,18 +82,49 @@ public override bool Execute()
 
         public async Task<bool> ExecuteAsync()
         {
+            if (!string.IsNullOrEmpty(ConnectionString))
```
Comment (chcosta) [inline src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs]: It's unfortunate to see this code duplicated so often.  Perhaps consider creating a ConnectionStringBasedTask class which is derived from Task and contains some of the shared logic.

File: src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs [hunk 2]
```diff
@@ -77,18 +82,49 @@ public override bool Execute()
...
+            {
+                if (!(string.IsNullOrEmpty(AccountKey) && string.IsNullOrEmpty(AccountName)))
+                {
+                    Log.LogError("If the ConnectionString property is set, you must not provide AccountKey / Name.  These values will be deprecated in the future.");
+                    return false;
```
Comment (chcosta) [inline src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs]: I know we do this everywhere, but the preferred method is to return `!Log.HasLoggedErrors`

**Pull request merged**